File: MiniDOMWebCtrls/DMD/Control/ControlPanelUserInfo.cs
using System;
using DMD;
using DMD.XML;
using minidom;
using static minidom.Sistema;

namespace minidom.Forms
{

    [Serializable]
    public class ControlPanelUserInfo 
        : IDMDXMLSerializable
    {

        [NonSerialized] public Sistema.CUser User;
        [NonSerialized] public Sistema.CLoginHistory LastLogin;
        // Public Attivita As CContattoUtente
        // Public DescrizioneAttivita As String

        public ControlPanelUserInfo()
        {
            DMDObject.IncreaseCounter(this);
        }

        public ControlPanelUserInfo(Sistema.CUser user) : this()
        {
            User = user;
        }

        public void Load()
        {
            LastLogin = User.CurrentLogin;
        }

        public void SetFieldInternal(string fieldName, object fieldValue)
        {
            switch (fieldName ?? "")
            {
                case "User":
                    {
                        User = (Sistema.CUser)fieldValue;
                        break;
                    }

                case "LastLogin":
                    {
                        LastLogin = null;
                        if (fieldValue is Sistema.CLoginHistory)
                            LastLogin = (Sistema.CLoginHistory)fieldValue;
                        break;
                    }
            }
        }

        public void XMLSerialize(XMLWriter writer)
        {
            writer.WriteTag("User", User);
            writer.WriteTag("LastLogin", LastLogin);
        }

        ~ControlPanelUserInfo()
        {
            DMDObject.DecreaseCounter(this);
        }
    }
}

File: MiniDOMWebCtrls/DMD/Control/ControlPanelUserInfo.vb
Imports Microsoft.VisualBasic
Imports minidom
Imports minidom.WebSite
Imports minidom.Sistema
Imports minidom.Databases
Imports minidom.CustomerCalls

Namespace Forms


    Public Class ControlPanelUserInfo
        Implements minidom.XML.IDMDXMLSerializable

        Public User As CUser
        Public LastLogin As CLoginHistory
        'Public Attivita As CContattoUtente
        'Public DescrizioneAttivita As String

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
        End Sub

        Public Sub New(ByVal user As CUser)
            Me.New
            Me.User = user
        End Sub

        Public Sub Load()
            Me.LastLogin = Me.User.CurrentLogin
        End Sub

        Public Sub SetFieldInternal(fieldName As String, fieldValue As Object) Implements XML.IDMDXMLSerializable.SetFieldInternal
            Select Case fieldName
                Case "User" : Me.User = fieldValue
                Case "LastLogin"
                    Me.LastLogin = Nothing
                    If (TypeOf (fieldValue) Is CLoginHistory) Then Me.LastLogin = fieldValue
            End Select
        End Sub

        Public Sub XMLSerialize(writer As XML.XMLWriter) Implements XML.IDMDXMLSerializable.XMLSerialize
            writer.WriteTag("User", Me.User)
            writer.WriteTag("LastLogin", Me.LastLogin)
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class


End Namespace